packages/wasi/src/index.ts

```diff
@@ -711,6 +799,26 @@ export default class WASIDefault {
           nwritten: number
         ) => {
           const stats = CHECK_FD(fd, WASI_RIGHT_FD_WRITE | WASI_RIGHT_FD_SEEK);
+          if (this.asyncified) {
+            return asyncify(async () => {
+              let written = 0;
+              for (const iov of getiovs(iovs, iovsLen)) {
+                let w = 0;
+                while (w < iov.byteLength) {
+                  w += await promisify(fs.write)(
```

Comment:
I think it would be better to use the `writeSync` API rather than make it awaitable and wait for the result.

```suggestion
                  let rr: number = fs.writeSync(
```